common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalBlockStoreClient.java

```diff
@@ -193,8 +194,14 @@ public void onFailure(Throwable e) {
         }
       });
     } catch (Exception e) {
-      logger.error("Exception while sending finalizeShuffleMerge request to {}:{}",
-        host, port, e);
+      if (e instanceof BlockPushNonFatalFailure) {
```

Comment:
Use `PUSH_ERROR_HANDLER#shouldLogError` instead.

Reply:
Also, I don't think we would catch this exception on the client side.
The exception is never thrown from the client side, and the client side would only get it in `callback#onFailure` which in turn passes it to the `MergeFinalizerListener`.
So, if you want to handle logging, you should handle it in the listener.